src/main/java/com/redhat/labs/omp/service/EngagementService.java

```diff
@@ -1,43 +1,258 @@
 package com.redhat.labs.omp.service;
 
+import java.util.List;
+import java.util.Optional;
+
 import javax.enterprise.context.ApplicationScoped;
 import javax.inject.Inject;
+import javax.json.bind.Jsonb;
 import javax.ws.rs.core.Response;
-import javax.ws.rs.core.UriBuilder;
-import javax.ws.rs.core.UriInfo;
 
-import org.eclipse.microprofile.rest.client.inject.RestClient;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-import com.redhat.labs.omp.cache.EngagementDataCache;
+import com.redhat.labs.omp.exception.ResourceAlreadyExistsException;
+import com.redhat.labs.omp.exception.ResourceNotFoundException;
 import com.redhat.labs.omp.model.Engagement;
-import com.redhat.labs.omp.rest.client.OMPGitLabAPIService;
+import com.redhat.labs.omp.model.git.api.FileAction;
+import com.redhat.labs.omp.model.git.api.GitApiEngagement;
+import com.redhat.labs.omp.model.git.api.GitApiFile;
+import com.redhat.labs.omp.repository.EngagementRepository;
 
 @ApplicationScoped
 public class EngagementService {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(EngagementService.class);
+
+    @ConfigProperty(name = "engagement.file.name", defaultValue = "engagement.json")
+    String engagementFileName;
```

Comment:
Git API representation for Engagement has been removed.  The mongo id has been changed to 'mongoId'.  When passed to the git-api, it will be ignored as there is no 'mongoId' in that Engagement object, which means it never makes it to the engagement.json file.  Finally, the project id has been changed to 'projectId' to match the PR for the git-api.